src/node.cc

```diff
@@ -393,6 +372,27 @@ MaybeLocal<Value> StartExecution(Environment* env, const char* main_script_id) {
           ->GetFunction(env->context())
           .ToLocalChecked()};
 
+  MaybeLocal<Value> result;
+  auto thread_switch_id =
+      env->is_main_thread() ? "internal/bootstrap/switches/is_main_thread"
+                            : "internal/bootstrap/switches/is_not_main_thread";
+  result = ExecuteBootstrapper(env, thread_switch_id, &parameters, &arguments);
+
+  if (result.IsEmpty()) {
+    return scope.EscapeMaybe(result);
+  }
+
+  auto process_state_switch_id =
+      env->owns_process_state()
+          ? "internal/bootstrap/switches/does_own_process_state"
+          : "internal/bootstrap/switches/does_not_own_process_state";
+  result = ExecuteBootstrapper(
+      env, process_state_switch_id, &parameters, &arguments);
+
+  if (result.IsEmpty()) {
+    return scope.EscapeMaybe(result);
+  }
```

Comment:
Just for context, are you moving this because you want to use the same snapshot for instances with different options?

Reply:
Yes (this would otherwise affect workers)

Reply:
hm, looks like the ordering of the worker makes sharing the snapshot just difficult (stdout and friends are set up only in these switches, but with the current error flow of workers they could need them to communicate with the parent when throwing `ERR_WORKER_INIT_FAILED` at bootstrap), so I'll leave them here now.